src/stash/src/upgrade/v35_to_v36.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+
+use crate::objects::{wire_compatible, WireCompatible};
+use crate::upgrade::MigrationAction;
+use crate::{StashError, Transaction, TypedCollection};
+
+pub mod v35 {
+    include!(concat!(env!("OUT_DIR"), "/objects_v35.rs"));
+}
+
+pub mod v36 {
+    include!(concat!(env!("OUT_DIR"), "/objects_v36.rs"));
+}
+
+wire_compatible!(v35::ServerConfigurationKey with v36::ServerConfigurationKey);
+wire_compatible!(v35::ServerConfigurationValue with v36::ServerConfigurationValue);
+
+const SYSTEM_CONFIGURATION_COLLECTION: TypedCollection<
+    v35::ServerConfigurationKey,
+    v35::ServerConfigurationValue,
+> = TypedCollection::new("system_configuration");
+
+const ENABLE_LD_RBAC_CHECKS_NAME: &str = "enable_ld_rbac_checks";
+const ENABLE_RBAC_CHECKS_NAME: &str = "enable_rbac_checks";
+
+/// Persist `false` for existing environments' RBAC flags, iff they're not already set.
+pub async fn upgrade(tx: &mut Transaction<'_>) -> Result<(), StashError> {
+    SYSTEM_CONFIGURATION_COLLECTION
+        .migrate_to(
+            tx,
+            |entries: &BTreeMap<v35::ServerConfigurationKey, v35::ServerConfigurationValue>| {
+                let mut updates = Vec::with_capacity(entries.len());
+                let mut found_ld_flag = false;
+                let mut found_user_flag = false;
+
+                for (key, value) in entries {
+                    if key.name == ENABLE_LD_RBAC_CHECKS_NAME {
+                        found_ld_flag = true;
```

Comment:
Just a heads up I think this is true for all environments, since we're the [default rule for the flag](https://app.launchdarkly.com/default/production/features/enable_ld_rbac_checks/targeting) is `true`